qemu/tests/balloon_check.py

```diff
@@ -215,7 +219,10 @@ def _memory_check_after_sub_test():
             try:
                 output = self.memory_check("after subtest", ballooned_mem)
             except error.TestFail:
-                return None
+                if self.small_size_mmem < output[0]:
+                    raise error.TestFail("Small Size Test failed")
```

Comment:
As suggested, we can start using `test.error()` here and elsewhere.
In this it would be `self.test.error(...)`